lib/ecto/query/builder.ex

```diff
@@ -682,6 +689,8 @@ defmodule Ecto.Query.Builder do
     do: type
   def validate_type!({:__aliases__, _, _} = type, _vars, env),
     do: Macro.expand(type, get_env(env))
+  def validate_type!({:parameterized, _, _} = type, _vars, _env),
+    do: Macro.escape(type)
```

Comment:
In a second thought, I should've escaped this **before** the validate_type!/3 call, right?